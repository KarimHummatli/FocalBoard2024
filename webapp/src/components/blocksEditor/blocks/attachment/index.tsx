import React, {useRef, useEffect} from 'react'
import {BlockInputProps, ContentType} from '../types'

import './attachment.scss'

const Attachment: ContentType = {
    name: 'attachment',
    displayName: 'Attachment',
    slashCommand: '/attachment',
    prefix: '',
    runSlashCommand: (): void => {},
    editable: false,
    Display: (props: BlockInputProps) => <div className='AttachmentView'>ðŸ“Ž {props.value}</div>,
    Input: (props: BlockInputProps) => {
        const ref = useRef<HTMLInputElement|null>(null)
        useEffect(() => {
            ref.current?.click()
        }, [])

        return (
            <input
                ref={ref}
                className='Attachment'
                type='file'
                onChange={(e) => {
                    const files = e.currentTarget?.files
                    if (files) {
                        for (let i = 0; i < files.length; i++) {
                            const file = files.item(i)
                            if (file) {
                                props.onSave(file.name as string)
                            }
                        }
                    }
                }}
            />
        )
    }
}

Attachment.runSlashCommand = (changeType: (contentType: ContentType) => void, changeValue: (value: string) => void): void => {
    changeType(Attachment)
    changeValue('')
}

export default Attachment
