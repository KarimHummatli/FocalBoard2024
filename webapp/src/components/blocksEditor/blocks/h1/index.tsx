import React, {useRef, useEffect} from 'react'
import {marked} from 'marked'

import {BlockInputProps, ContentType} from '../types'

import './h1.scss'

const H1: ContentType = {
    name: 'h1',
    displayName: 'Title',
    slashCommand: '/title',
    prefix: '# ',
    render: (value: string) => {
        const renderer = new marked.Renderer()
        const html = marked('# '+value, {renderer, breaks: true})
        return <div
            dangerouslySetInnerHTML={{__html: html.trim()}}
        />
    },
    runSlashCommand: (): void => {},
    Input: (props: BlockInputProps) => {
        const ref = useRef<HTMLInputElement|null>(null)
        useEffect(() => {
            ref.current?.focus()
        }, [])
        return (
            <input
                ref={ref}
                className='H1'
                onChange={(e) => props.onChange(e.currentTarget.value)}
                onKeyDown={(e) => {
                    if (props.value === '' && e.key === "Backspace") {
                        props.onCancel()
                    }
                    if (e.key === "Enter") {
                        props.onSave(props.value)
                    }
                }}
                value={props.value}
            />
        )
    }
}

H1.runSlashCommand = (changeType: (contentType: ContentType) => void, changeValue: (value: string) => void, ...args: string[]): void => {
    changeType(H1)
    changeValue(args.join(' '))
}

export default H1
