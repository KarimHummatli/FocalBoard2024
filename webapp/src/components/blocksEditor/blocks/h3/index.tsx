import React, {useRef, useEffect} from 'react'
import {marked} from 'marked'

import {BlockInputProps, ContentType} from '../types'

import './h3.scss'

const H3: ContentType = {
    name: 'h3',
    displayName: 'Sub Sub title',
    slashCommand: '/subsubtitle',
    prefix: '### ',
    render: (value: string) => {
        const renderer = new marked.Renderer()
        const html = marked('### '+value, {renderer, breaks: true})
        return <div
            dangerouslySetInnerHTML={{__html: html.trim()}}
        />
    },
    runSlashCommand: (): void => {},
    Input: (props: BlockInputProps) => {
        const ref = useRef<HTMLInputElement|null>(null)
        useEffect(() => {
            ref.current?.focus()
        }, [])
        return (
            <input
                ref={ref}
                className='H3'
                onChange={(e) => props.onChange(e.currentTarget.value)}
                onKeyDown={(e) => {
                    if (props.value === '' && e.key === "Backspace") {
                        props.onCancel()
                    }
                    if (e.key === "Enter") {
                        props.onSave(props.value)
                    }
                }}
                value={props.value}
            />
        )
    }
}

H3.runSlashCommand = (changeType: (contentType: ContentType) => void, changeValue: (value: string) => void, ...args: string[]): void => {
    changeType(H3)
    changeValue(args.join(' '))
}

export default H3
