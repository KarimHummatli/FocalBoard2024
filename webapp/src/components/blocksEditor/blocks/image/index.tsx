import React, {useRef, useEffect} from 'react'
import {BlockInputProps, ContentType} from '../types'

import './image.scss'

type FileInfo = {
    file: string|ArrayBuffer
    width?: number
    align?: 'left'|'center'|'right'
}

const Image: ContentType<FileInfo> = {
    name: 'image',
    displayName: 'Image',
    slashCommand: '/image',
    prefix: '',
    runSlashCommand: (): void => {},
    editable: false,
    Display: (props: BlockInputProps<FileInfo>) => {
        if (props.value.file) {
            return <img className='ImageView' src={props.value.file as string}/>
        }
        return null
    },
    Input: (props: BlockInputProps<FileInfo>) => {
        const ref = useRef<HTMLInputElement|null>(null)
        useEffect(() => {
            ref.current?.click()
        }, [])

        return (
            <div>
                {props.value.file && (typeof props.value.file === 'string') && (
                    <img
                        className='ImageView' src={props.value.file}
                        onClick={() => ref.current?.click()}
                    />
                )}
                <input
                    ref={ref}
                    className='Image'
                    type='file'
                    accept='image/*'
                    onChange={(e) => {
                        const reader = new FileReader();
                        const file = (e.currentTarget?.files || [])[0]
                        reader.readAsArrayBuffer(file);
                        reader.onload = function () {
                            props.onSave({file: reader.result as ArrayBuffer})
                        }
                    }}
                />
            </div>
        )
    }
}

Image.runSlashCommand = (changeType: (contentType: ContentType<FileInfo>) => void, changeValue: (value: FileInfo) => void): void => {
    changeType(Image)
    changeValue({file: ''})
}

export default Image
