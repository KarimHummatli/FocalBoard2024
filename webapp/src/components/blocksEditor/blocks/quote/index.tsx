import React, {useRef, useEffect} from 'react'
import {marked} from 'marked'

import {BlockInputProps, ContentType} from '../types'

import './quote.scss'

const Quote: ContentType = {
    name: 'quote',
    displayName: 'Quote',
    slashCommand: '/quote',
    prefix: '> ',
    render: (value: string) => {
        const renderer = new marked.Renderer()
        const html = marked('> '+value, {renderer, breaks: true})
        return <div
            className='Quote'
            dangerouslySetInnerHTML={{__html: html.trim()}}
        />
    },
    runSlashCommand: (): void => {},
    Input: (props: BlockInputProps) => {
        const ref = useRef<HTMLInputElement|null>(null)
        useEffect(() => {
            ref.current?.focus()
        }, [])
        return (
            <blockquote className='Quote'>
                <input
                    ref={ref}
                    onChange={(e) => props.onChange(e.currentTarget.value)}
                    onKeyDown={(e) => {
                        if (props.value === '' && e.key === "Backspace") {
                            props.onCancel()
                        }
                        if (e.key === "Enter") {
                            props.onSave(props.value)
                        }
                    }}
                    onBlur={() => props.onSave(props.value)}
                    value={props.value}
                />
            </blockquote>
        )
    }
}

Quote.runSlashCommand = (changeType: (contentType: ContentType) => void, changeValue: (value: string) => void, ...args: string[]): void => {
    changeType(Quote)
    changeValue(args.join(' '))
}

export default Quote
