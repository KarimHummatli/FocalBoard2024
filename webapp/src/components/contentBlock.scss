.ContentBlock {
    .MenuWrapper,
    .dnd-handle {
        display: none;
    }
    &:hover {
        .MenuWrapper {
            display: flex;
        }
        .dnd-handle {
            display: flex;
            opacity: 0.8;
            height: 18px;
        }
    }
    > * {
        flex: 1 1 auto;
    }
    > .octo-block-margin {
        flex: 0 0 auto;
    }
    .ImageElement {
        pointer-events: none;
    }
    .octo-block {
        width: 85%;
    }
}

.rowContents {
    display: flex;
    width: 100%;
}

.addToRow {
    background-color: yellow;
    width: 10px;
}