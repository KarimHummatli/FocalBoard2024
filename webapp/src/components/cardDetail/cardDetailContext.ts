// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react'

import {Card} from '../../blocks/card'
import {ContentHandler} from '../content/contentRegistry'

export type CardDetailContextType = {
    card: Card
    newBlockId: string
    resetNewBlockId: () => void
    addNewBlock: (handler: ContentHandler, index: number) => void
}

const CardDetailContext = React.createContext({} as CardDetailContextType)

export default CardDetailContext
