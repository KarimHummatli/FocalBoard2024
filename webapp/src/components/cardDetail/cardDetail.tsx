// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React, {useCallback, useEffect, useRef, useState, Fragment, useMemo} from 'react'
import {FormattedMessage, useIntl, IntlShape} from 'react-intl'

import {BlockIcons} from '../../blockIcons'
import {Card} from '../../blocks/card'
import {BoardView} from '../../blocks/boardView'
import {Board} from '../../blocks/board'
import {CommentBlock} from '../../blocks/commentBlock'
import {ContentBlock} from '../../blocks/contentBlock'
import {Block, ContentBlockTypes, createBlock} from '../../blocks/block'
import mutator from '../../mutator'
import octoClient from '../../octoClient'
import Button from '../../widgets/buttons/button'
import {Focusable} from '../../widgets/editable'
import EditableArea from '../../widgets/editableArea'
import CompassIcon from '../../widgets/icons/compassIcon'
import TelemetryClient, {TelemetryActions, TelemetryCategory} from '../../telemetry/telemetryClient'

import BlockIconSelector from '../blockIconSelector'

import {useAppDispatch} from '../../store/hooks'
import {setCurrent as setCurrentCard} from '../../store/cards'
import {Permission} from '../../constants'
import {useHasCurrentBoardPermissions} from '../../hooks/permissions'
import BlocksEditor from '../blocksEditor/blocksEditor'
import {BlockData} from '../blocksEditor/blocks/types'

import CardSkeleton from '../../svg/card-skeleton'

import CommentsList from './commentsList'
import CardDetailProperties from './cardDetailProperties'
import useImagePaste from './imagePaste'

import './cardDetail.scss'

export const OnboardingBoardTitle = 'Welcome to Boards!'
export const OnboardingCardTitle = 'Create a new card'

type Props = {
    board: Board
    activeView: BoardView
    views: BoardView[]
    cards: Card[]
    card: Card
    comments: CommentBlock[]
    contents: Array<ContentBlock|ContentBlock[]>
    readonly: boolean
    onClose: () => void
}

async function addBlock(card: Card, intl: IntlShape, title: string, fields: any, contentType: ContentBlockTypes, afterBlockId: string): Promise<Block> {
    const block = createBlock()
    block.parentId = card.id
    block.boardId = card.boardId
    block.title = title
    block.type = contentType
    block.fields = {...block.fields, ...fields}

    const description = intl.formatMessage({id: 'CardDetail.addCardText', defaultMessage: 'add card text'})

    const afterRedo = async (newBlock: Block) => {
        const contentOrder = card.fields.contentOrder.slice()
        if (afterBlockId) {
            const idx = contentOrder.indexOf(afterBlockId)
            if (idx === -1) {
                contentOrder.push(newBlock.id)
            } else {
                contentOrder.splice(idx+1, 0, newBlock.id)
            }
        } else {
            contentOrder.push(newBlock.id)
        }
        await octoClient.patchBlock(card.boardId, card.id, {updatedFields: {contentOrder}})
    }

    const beforeUndo = async () => {
        const contentOrder = card.fields.contentOrder.slice()
        await octoClient.patchBlock(card.boardId, card.id, {updatedFields: {contentOrder}})
    }

    return mutator.insertBlock(block.boardId, block, description, afterRedo, beforeUndo)
}

function moveBlock(boardId: string, blockId: string, dstBlockId: string): void {
    // TODO: Make this a mutation
    octoClient.moveBlockTo(boardId, blockId, "after", dstBlockId)
}


const CardDetail = (props: Props): JSX.Element|null => {
    const {card, comments} = props
    const {limited} = card
    const [title, setTitle] = useState(card.title)
    const [serverTitle, setServerTitle] = useState(card.title)
    const titleRef = useRef<Focusable>(null)
    const saveTitle = useCallback(() => {
        if (title !== card.title) {
            mutator.changeBlockTitle(props.board.id, card.id, card.title, title)
        }
    }, [card.title, title])
    const canEditBoardCards = useHasCurrentBoardPermissions([Permission.ManageBoardCards])
    const canCommentBoardCards = useHasCurrentBoardPermissions([Permission.CommentBoardCards])

    const saveTitleRef = useRef<() => void>(saveTitle)
    saveTitleRef.current = saveTitle
    const intl = useIntl()

    useImagePaste(props.board.id, card.id, card.fields.contentOrder)

    useEffect(() => {
        if (!title) {
            setTimeout(() => titleRef.current?.focus(), 300)
        }
        TelemetryClient.trackEvent(TelemetryCategory, TelemetryActions.ViewCard, {board: props.board.id, view: props.activeView.id, card: card.id})
    }, [])

    useEffect(() => {
        if (serverTitle === title) {
            setTitle(card.title)
        }
        setServerTitle(card.title)
    }, [card.title, title])

    useEffect(() => {
        return () => {
            saveTitleRef.current && saveTitleRef.current()
        }
    }, [])

    const setRandomIcon = useCallback(() => {
        const newIcon = BlockIcons.shared.randomIcon()
        mutator.changeBlockIcon(props.board.id, card.id, card.fields.icon, newIcon)
    }, [card.id, card.fields.icon])

    const dispatch = useAppDispatch()
    useEffect(() => {
        dispatch(setCurrentCard(card.id))
    }, [card.id])

    if (!card) {
        return null
    }

    console.log(props.contents)
    const blocks = useMemo(() => props.contents.flatMap((value: Block | Block[]): BlockData<any> => {
        let v: Block = Array.isArray(value) ? value[0] : value

        let data: any = v?.title
        if (v?.type === 'image') {
            data = v?.fields.fileId
        }

        if (v?.type === 'checkbox') {
            data = {
                value: v?.title,
                checked: v?.fields.value,
            }
        }

        return {
            id: v?.id,
            value: data,
            contentType: v?.type,
        }
    }), [props.contents])
    console.log(blocks)

    return (
        <>
            <div className={`CardDetail content${limited ? ' is-limited' : ''}`}>
                <BlockIconSelector
                    block={card}
                    size='l'
                    readonly={props.readonly || !canEditBoardCards || limited}
                />
                {!props.readonly && canEditBoardCards && !card.fields.icon &&
                    <div className='add-buttons'>
                        <Button
                            emphasis='default'
                            size='small'
                            onClick={setRandomIcon}
                            icon={
                                <CompassIcon
                                    icon='emoticon-outline'
                                />}

                        >
                            <FormattedMessage
                                id='CardDetail.add-icon'
                                defaultMessage='Add icon'
                            />
                        </Button>
                    </div>}

                <EditableArea
                    ref={titleRef}
                    className='title'
                    value={title}
                    placeholderText='Untitled'
                    onChange={(newTitle: string) => setTitle(newTitle)}
                    saveOnEsc={true}
                    onSave={saveTitle}
                    onCancel={() => setTitle(props.card.title)}
                    readonly={props.readonly || !canEditBoardCards || limited}
                    spellCheck={true}
                />

                {/* Hidden (limited) card copy + CTA */}

                {limited && <div className='CardDetail__limited-wrapper'>
                    <CardSkeleton
                        className='CardDetail__limited-bg'
                    />
                    <p className='CardDetail__limited-title'>
                        <FormattedMessage
                            id='CardDetail.limited-title'
                            defaultMessage='This card is hidden'
                        />
                    </p>
                    <p className='CardDetail__limited-body'>
                        <FormattedMessage
                            id='CardDetail.limited-body'
                            defaultMessage='Upgrade to our Professional or Enterprise plan to view archived cards, have unlimited views per boards, unlimited cards and more.'
                        />
                        <br/>
                        <a
                            className='CardDetail__limited-link'
                            role='button'
                            onClick={() => {
                                props.onClose();
                                (window as any).openPricingModal()({trackingLocation: 'boards > learn_more_about_our_plans_click'})
                            }}
                        >
                            <FormattedMessage
                                id='CardDetial.limited-link'
                                defaultMessage='Learn more about our plans.'
                            />
                        </a>
                    </p>
                    <Button
                        className='CardDetail__limited-button'
                        onClick={() => {
                            props.onClose();
                            (window as any).openPricingModal()({trackingLocation: 'boards > upgrade_click'})
                        }}
                        emphasis='primary'
                        size='large'
                    >
                        {intl.formatMessage({id: 'CardDetail.limited-button', defaultMessage: 'Upgrade'})}
                    </Button>
                </div>}

                {/* Property list */}

                {!limited &&
                <CardDetailProperties
                    board={props.board}
                    card={props.card}
                    cards={props.cards}
                    activeView={props.activeView}
                    views={props.views}
                    readonly={props.readonly}
                />}

                {/* Comments */}

                {!limited && <Fragment>
                    <hr/>
                    <CommentsList
                        comments={comments}
                        boardId={card.boardId}
                        cardId={card.id}
                        readonly={props.readonly || !canCommentBoardCards}
                    />
                </Fragment>}
            </div>

            {/* Content blocks */}

            {!limited && <div className='CardDetail content fullwidth content-blocks'>
                <BlocksEditor
                    blocks={blocks}
                    onBlockCreated={(block: any, afterBlock: any): any => {
                        if (block.contentType === 'text' && block.value === '') {
                            return null
                        }
                        if (block.contentType === 'checkbox') {
                            addBlock(card, intl, block.value.value, {value: block.value.checked}, block.contentType, afterBlock?.id)
                        } else {
                            addBlock(card, intl, block.value, {}, block.contentType, afterBlock?.id)
                        }
                        return block
                    }}
                    onBlockModified={(block: any): BlockData<any>|null => {
                        const originalContentBlock = props.contents.flatMap((b) => b).find((b) => b.id === block.id)
                        if (!originalContentBlock) {
                            return null
                        }

                        if (block.contentType === 'text' && block.value === '') {
                            const description = intl.formatMessage({id: 'ContentBlock.DeleteAction', defaultMessage: 'delete'})

                            mutator.deleteBlock(originalContentBlock, description)
                            return null
                        }
                        const newBlock = {
                            ...originalContentBlock,
                            title: block.value,
                        }

                        if (block.contentType === 'checkbox') {
                            newBlock.title = block.value.value
                            newBlock.fields = {...newBlock.fields, value: block.value.checked}
                        }
                        mutator.updateBlock(card.boardId, newBlock, originalContentBlock, intl.formatMessage({id: 'ContentBlock.editCardText', defaultMessage: 'edit card content'}))
                        return block
                    }}
                    onBlockMoved={(block: any, afterBlock: any) => {
                        console.log(block, afterBlock)
                        moveBlock(card.boardId, block.id, afterBlock.id)
                    }}
                />
            </div>}
        </>
    )
}
                // <CardDetailProvider card={card}>
                //     <CardDetailContents
                //         card={props.card}
                //         contents={props.contents}
                //         readonly={props.readonly || !canEditBoardCards}
                //     />
                //     {!props.readonly && canEditBoardCards && <CardDetailContentsMenu/>}
                // </CardDetailProvider>

export default CardDetail
