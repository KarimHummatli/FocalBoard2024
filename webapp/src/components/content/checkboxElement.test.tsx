// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {ReactElement, ReactNode} from 'react'
import {fireEvent, render} from '@testing-library/react'
import '@testing-library/jest-dom'

import {wrapIntl} from '../../testUtils'

import {ContentBlock} from '../../blocks/contentBlock'

import {CardDetailProvider} from '../cardDetail/cardDetailContext'

import {TestBlockFactory} from '../../test/testBlockFactory'

import CheckboxElement from './checkboxElement'

const fetchMock = require('fetch-mock-jest')

const board = TestBlockFactory.createBoard()
const card = TestBlockFactory.createCard(board)

const wrap = (child: ReactNode): ReactElement => (
    wrapIntl(
        <CardDetailProvider card={card}>
            {child}
        </CardDetailProvider>,
    )
)

describe('components/content/CheckboxElement', () => {
    const defaultBlock: ContentBlock = {
        id: 'test-id',
        workspaceId: '',
        parentId: '',
        rootId: '',
        modifiedBy: 'test-user-id',
        schema: 0,
        type: 'checkbox',
        title: 'test-title',
        fields: {},
        createdBy: 'test-user-id',
        createAt: 0,
        updateAt: 0,
        deleteAt: 0,
    }

    beforeAll(() => {
        fetchMock.post('*', {})
    })

    afterAll(() => {
        fetchMock.mockClear()
    })

    test('should match snapshot', () => {
        const component = wrap(
            <CheckboxElement
                block={defaultBlock}
                readonly={false}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot on read only', () => {
        const component = wrap(
            <CheckboxElement
                block={defaultBlock}
                readonly={true}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot on change title', () => {
        const component = wrap(
            <CheckboxElement
                block={defaultBlock}
                readonly={false}
            />,
        )
        const {container, getByTitle} = render(component)
        const input = getByTitle(/test-title/i)
        fireEvent.blur(input, {target: {textContent: 'changed name'}})
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot on toggle', () => {
        const component = wrap(
            <CheckboxElement
                block={defaultBlock}
                readonly={false}
            />,
        )
        const {container, getByRole} = render(component)
        const input = getByRole('checkbox')
        fireEvent.change(input, {target: {value: 'on'}})
        expect(container).toMatchSnapshot()
    })
})
