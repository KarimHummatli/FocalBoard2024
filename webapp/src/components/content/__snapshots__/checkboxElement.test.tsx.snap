// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/content/checkboxElement should change title 1`] = `
<div>
  <div
    class="CheckboxElement"
  >
    <input
      id="checkbox-test-id"
      type="checkbox"
      value="off"
    />
    <div
      class="EditableAreaWrap"
    >
      <textarea
        class="EditableArea Editable "
        height="0"
        placeholder="Edit text..."
        rows="1"
        spellcheck="true"
        title="new title"
      >
        new title
      </textarea>
      <div
        class="EditableAreaContainer"
      >
        <textarea
          aria-hidden="true"
          class="EditableAreaReference Editable "
          dir="auto"
          disabled=""
          rows="1"
        >
          new title
        </textarea>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/content/checkboxElement should match snapshot 1`] = `
<div>
  <div
    class="CheckboxElement"
  >
    <input
      id="checkbox-test-id"
      type="checkbox"
      value="off"
    />
    <div
      class="EditableAreaWrap"
    >
      <textarea
        class="EditableArea Editable "
        height="0"
        placeholder="Edit text..."
        rows="1"
        spellcheck="true"
        title="test-title"
      >
        test-title
      </textarea>
      <div
        class="EditableAreaContainer"
      >
        <textarea
          aria-hidden="true"
          class="EditableAreaReference Editable "
          dir="auto"
          disabled=""
          rows="1"
        >
          test-title
        </textarea>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/content/checkboxElement should match snapshot when read only 1`] = `
<div>
  <div
    class="CheckboxElement"
  >
    <input
      disabled=""
      id="checkbox-test-id"
      type="checkbox"
      value="off"
    />
    <div
      class="EditableAreaWrap"
    >
      <textarea
        class="EditableArea Editable readonly "
        height="0"
        placeholder="Edit text..."
        readonly=""
        rows="1"
        spellcheck="true"
        title="test-title"
      >
        test-title
      </textarea>
      <div
        class="EditableAreaContainer"
      >
        <textarea
          aria-hidden="true"
          class="EditableAreaReference Editable readonly "
          dir="auto"
          disabled=""
          rows="1"
        >
          test-title
        </textarea>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/content/checkboxElement should toggle value 1`] = `
<div>
  <div
    class="CheckboxElement"
  >
    <input
      id="checkbox-test-id"
      type="checkbox"
      value="on"
    />
    <div
      class="EditableAreaWrap"
    >
      <textarea
        class="EditableArea Editable "
        height="0"
        placeholder="Edit text..."
        rows="1"
        spellcheck="true"
        title="test-title"
      >
        test-title
      </textarea>
      <div
        class="EditableAreaContainer"
      >
        <textarea
          aria-hidden="true"
          class="EditableAreaReference Editable "
          dir="auto"
          disabled=""
          rows="1"
        >
          test-title
        </textarea>
      </div>
    </div>
  </div>
</div>
`;
